sdk/python/kfp/v2/dsl/component_bridge.py

```diff
@@ -0,0 +1,201 @@
+          input_name].runtime_value.constant_value.string_value = argument_value
+    elif isinstance(argument_value, int):
+      pipeline_task_spec.inputs.parameters[
+          input_name].runtime_value.constant_value.int_value = argument_value
+    elif isinstance(argument_value, float):
+      pipeline_task_spec.inputs.parameters[
+          input_name].runtime_value.constant_value.double_value = argument_value
+    else:
+      raise NotImplementedError(
+          'Input argument supports only the following types: PipelineParam'
+          ', str, int, float. Got: "{}".'.format(argument_value))
+
+  for output in component_spec.outputs or []:
+    if type_utils.is_artifact_type(output.type):
+      pipeline_task_spec.outputs.artifacts[
+          output.name].artifact_type.instance_schema = (
+              type_utils.get_artifact_type_schema(output.type))
+    elif type_utils.is_parameter_type(output.type):
+      pipeline_task_spec.outputs.parameters[
+          output.name].type = type_utils.get_parameter_type(output.type)
+    else:
+      raise NotImplementedError(
+          'Unsupported output type: "{}". The type must be one of the following: {}.'
+          .format(output.type, type_utils.all_types()))
+
+  def _input_artifact_placeholder(input_key: str) -> str:
+    return "{{{{$.inputs.artifacts['{}'].uri}}}}".format(input_key)
+
+  def _input_parameter_placeholder(input_key: str) -> str:
+    return "{{{{$.inputs.parameters['{}']}}}}".format(input_key)
+
+  def _output_artifact_placeholder(input_key: str) -> str:
+    return "{{{{$.outputs.artifacts['{}'].uri}}}}".format(input_key)
+
+  # IR placeholders are decided merely based on the declared type of the input.
+  # It doesn't matter wether it's InputValuePlaceholder or InputPathPlaceholder
+  # from component_spec.
+  placeholder_arguments = {
+      input_spec.name: _input_artifact_placeholder(input_spec.name)
+      if type_utils.is_artifact_type(input_spec.type) else
+      _input_parameter_placeholder(input_spec.name)
+      for input_spec in component_spec.inputs or []
+  }
+
+  resolved_cmd_ir = _resolve_command_line_and_paths(
+      component_spec=component_spec,
+      arguments=placeholder_arguments,
+      input_path_generator=_input_artifact_placeholder,
+      output_path_generator=_output_artifact_placeholder,
+  )
+
+  resolved_cmd = _resolve_command_line_and_paths(
+      component_spec=component_spec,
+      arguments=arguments,
+  )
+
+  container_spec = component_spec.implementation.container
+
+  pipeline_container_spec = (
+      pipeline_spec_pb2.PipelineDeploymentConfig.PipelineContainerSpec())
+  pipeline_container_spec.image = container_spec.image
+  pipeline_container_spec.command.extend(resolved_cmd_ir.command)
+  pipeline_container_spec.args.extend(resolved_cmd_ir.args)
+
+  old_warn_value = dsl.ContainerOp._DISABLE_REUSABLE_COMPONENT_WARNING
+  dsl.ContainerOp._DISABLE_REUSABLE_COMPONENT_WARNING = True
+  task = dsl.ContainerOp(
```

Comment:
Do we still need to carry the ContainerOp in the v2 logic?

Reply:
FAIK, the DSL is not being changed in the initial version.
To build a pipeline, there needs to exist a Task-like object (an object that has `.outputs` dictionary).
Currently the `dsl.ContainerOp` is that object.

Reply:
I see. You're right. The `PipelineCOntainerSpec` does not suffice